### PR TITLE
feat: added frozen collection article

### PR DESCRIPTION
**Pull request by linkdotnet**
This PR adds a topic for .NET 8, which will land somewhen November 2023.

### CODE REVIEW

File: drafts/frozen-collections.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+title: Blazor Project Structure
+domain: campzulu.hashnode.dev
+tags: .NET, C#, .NET 8
```
Comment (lehmamic) [inline drafts/frozen-collections.md:4]: I'm not sure whether Hashnode can handle unknown Tags or not, maybe we just need to try it out ^^

File: drafts/frozen-collections.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    public void LookupImmutableHashSet()
+    {
+        for (var i = 0; i < Iterations; i++)
+            _ = immutableHashSet.Contains(i);
+    }
+}
+```
+
+And here are the results:
+```no-class
```
Comment (lehmamic) [inline drafts/frozen-collections.md:107]: ha, I just learned something ^^

File: drafts/frozen-collections.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+|                 Method |         Mean |     Error |    StdDev |  Ratio | RatioSD |
+|----------------------- |-------------:|----------:|----------:|-------:|--------:|
+|             CreateList |     251.9 ns |   2.41 ns |   2.14 ns |   1.00 |    0.00 |
+|       CreateFrozenList |  57,626.2 ns | 973.67 ns | 910.77 ns | 228.94 |    4.17 |
+|          CreateHashSet |   6,694.7 ns | 132.08 ns | 171.74 ns |  26.72 |    0.74 |
+| CreateImmutableHashSet | 200,541.6 ns | 972.71 ns | 812.25 ns | 796.58 |    8.33 |
+```
+
+## Conclusion
+The .NET Framework will get more and more specialized types which can help us to express our intent better and also help us with writing high-performant code. A new welcome entry to this is the **frozen** collections!
```
Comment (lehmamic) [inline drafts/frozen-collections.md:159]: Maybe you list your sources?

**Review by lehmamic — APPROVED** [review]
Generally a very good article. I added some questions, but basically, we can merge the pr.

File: drafts/frozen-collections.md (hunk 3 above)
Comment (linkdotnet) [reply drafts/frozen-collections.md:159]: The source is on line 127

File: drafts/frozen-collections.md (hunk 1 above)
Comment (r3dDoX) [reply drafts/frozen-collections.md:4]: Yes, changes should always be possible

**Pull request merged**